compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -14938,7 +14938,8 @@ private STNode parseTypedBindingPatternOrExprRhs(STNode typeOrExpr, boolean allo
                 }
 
                 // If the next token is an equal, then it is typedBP with union/intersection type and missing var name
-                if (peek().kind == SyntaxKind.EQUAL_TOKEN) {
+                if (peek().kind == SyntaxKind.EQUAL_TOKEN && 
+                        (!isDefiniteExpr(typeOrExpr.kind) && !isDefiniteExpr(rhsTypedBPOrExpr.kind))) {
```

Comment:
Do we need this change here? since we call,
```
lhsType = getTypeDescFromExpr(lhsType);
rhsType = getTypeDescFromExpr(rhsType);
```
inside `createCaptureBPWithMissingVarName()` ?

Reply:
removed